File: pingdartdb-go/test/test_sdk.go
package main

import (
	"bytes"
	"crypto/aes"
	"crypto/cipher"
	"crypto/rand"
	"crypto/sha256"
	"encoding/base64"
	"encoding/hex"
	"encoding/json"
	"fmt"
	"time"

	"github.com/pingdart/pingdartdb-go"
)

func generateValidKey() string {
	secretKey := "PingDartSuperSecretKey2026!@#$"
	hash := sha256.Sum256([]byte(secretKey))
	encodedHash := base64.StdEncoding.EncodeToString(hash[:])
	key := []byte(encodedHash[:32])

	iv := make([]byte, 16)
	rand.Read(iv)

	payload := map[string]string{"tier": "free", "createdAt": time.Now().Format(time.RFC3339)}
	jsonData, _ := json.Marshal(payload)

	// PKCS7 Padding
	padding := 16 - len(jsonData)%16
	padtext := bytes.Repeat([]byte{byte(padding)}, padding)
	jsonData = append(jsonData, padtext...)

	block, _ := aes.NewCipher(key)
	mode := cipher.NewCBCEncrypter(block, iv)

	encrypted := make([]byte, len(jsonData))
	mode.CryptBlocks(encrypted, jsonData)

	return "pd_" + hex.EncodeToString(iv) + "." + hex.EncodeToString(encrypted)
}

func main() {
	fmt.Println("1. Generated Local Go Offline API Key")
	apiKey := generateValidKey()

	config := pingdartdb.DBConfig{
		Host:     "94.136.190.60",
		User:     "pingdart_ravi",
		Password: "Ravindra12@",
		Database: "pingdart_test",
		Type:     "mysql",
	}

	db, err := pingdartdb.NewPingDartDB(apiKey, config)
	if err != nil {
		fmt.Printf("❌ Go Test failed: %v\n", err)
		return
	}

	err = db.Connect()
	if err != nil {
		fmt.Printf("❌ Go Test failed: %v\n", err)
		return
	}
	defer db.Close()
	fmt.Println("2. Connection successful.")

	options := map[string]interface{}{
		"conditions": map[string]interface{}{
			"status": "active",
		},
	}

	res, err := db.Table("test_users").Read(options)
	if err != nil {
		fmt.Printf("❌ Go Test failed during read: %v\n", err)
		return
	}

	fmt.Printf("3. Basic Read test returned %v rows.\n", res["totalCount"])
	fmt.Println("\n✅ Go SDK Connection Test passed successfully!")
}

File: pingdartdb-java/src/main/java/com/pingdart/TestSDK.java
package com.pingdart;

import java.time.Instant;
import java.util.Base64;
import java.util.HashMap;
import java.util.Map;
import java.security.SecureRandom;
import java.security.MessageDigest;
import java.nio.charset.StandardCharsets;
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import com.fasterxml.jackson.databind.ObjectMapper;

public class TestSDK {

    private static String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) sb.append(String.format("%02x", b));
        return sb.toString();
    }

    private static String generateValidKey() throws Exception {
        String secretKey = "PingDartSuperSecretKey2026!@#$";
        MessageDigest digest = MessageDigest.getInstance("SHA-256");
        byte[] hash = digest.digest(secretKey.getBytes(StandardCharsets.UTF_8));
        String encodedHash = Base64.getEncoder().encodeToString(hash);
        byte[] keyBytes = encodedHash.substring(0, 32).getBytes(StandardCharsets.UTF_8);

        byte[] iv = new byte[16];
        new SecureRandom().nextBytes(iv);

        Map<String, String> payload = new HashMap<>();
        payload.put("tier", "free");
        payload.put("createdAt", Instant.now().toString());

        ObjectMapper mapper = new ObjectMapper();
        byte[] jsonData = mapper.writeValueAsBytes(payload);

        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
        SecretKeySpec keySpec = new SecretKeySpec(keyBytes, "AES");
        IvParameterSpec ivSpec = new IvParameterSpec(iv);
        cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivSpec);

        byte[] encrypted = cipher.doFinal(jsonData);

        return "pd_" + bytesToHex(iv) + "." + bytesToHex(encrypted);
    }

    public static void main(String[] args) {
        try {
            System.out.println("1. Generated Local Java Offline API Key");
            String apiKey = generateValidKey();

            Map<String, String> config = new HashMap<>();
            config.put("host", "94.136.190.60");
            config.put("user", "pingdart_ravi");
            config.put("password", "Ravindra12@");
            config.put("database", "pingdart_test");
            config.put("type", "mysql");

            PingDartDB db = new PingDartDB(apiKey, config);
            db.connect();
            System.out.println("2. Connection successful.");

            Map<String, Object> options = new HashMap<>();
            Map<String, Object> conditions = new HashMap<>();
            conditions.put("status", "active");
            options.put("conditions", conditions);

            Map<String, Object> res = db.table("test_users").read(options);
            System.out.println("3. Basic Read test returned " + res.get("totalCount") + " rows.");

            db.close();
            System.out.println("\n✅ Java SDK Connection Test passed successfully!");

        } catch (Exception e) {
            System.err.println("❌ Java Test failed: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
